src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,73 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+import { TodoFilterTypes } from '../../types/TodoFilterTypes';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+type Props = {
+  query: string;
+  onQueryChange: (value: string) => void;
+  selectedFilter: string;
+  onSelectedFilter: (value: TodoFilterTypes) => void;
+};
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+export const TodoFilter: React.FC<Props> = ({
+  query,
+  onQueryChange,
+  selectedFilter,
+  onSelectedFilter,
+}) => {
+  const handleReset = () => {
+    onQueryChange('');
+    onSelectedFilter(TodoFilterTypes.All);
```

Comment:
But I guess better not to reset "Filter" on this 'x' icon click, because it's related only to search